### PR TITLE
One Terminal Script

### PR DESCRIPTION
**Pull request by mashrukk**
## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: tabs/applications-setup/tab_data.toml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 name = "Applications Setup"
 
 [[data]]
-name = "Alacritty"
-script = "alacritty-setup.sh"
+name = "Terminals"
```
Comment (adamperkowski) [inline tabs/applications-setup/tab_data.toml:4]:
```suggestion
name = "Terminal Emulators"
```

File: tabs/applications-setup/terminal-setup.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
+#!/bin/sh -e
+
+. ../common-script.sh
+
+chooseTerminal() {
+    echo "Choose your preferred terminal:"
```
Comment (adamperkowski) [inline tabs/applications-setup/terminal-setup.sh:6]:
```suggestion
    echo "Choose your preferred terminal emulator:"
```

File: tabs/applications-setup/terminal-setup.sh [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        1)
+            setupAlacritty
+            setupAlacrittyConfig
+            ;;
+        2)
+            setupKitty
+            setupKittyConfig
+            ;;
+        *)
+            printf "%b\n" "${RED}Invalid choice. Please choose one of the options available.${RC}" >&2
```
Comment (adamperkowski) [inline tabs/applications-setup/terminal-setup.sh:22]:
```suggestion
            printf "%b\n" "${RED}Invalid choice. Please choose one of the options available.${RC}"
```

**Review by adamperkowski — CHANGES_REQUESTED** [review]

**Comment by adamperkowski** [comment]
This would break some PRs. I'd say you should wait until #433 #367 and other ones that conflict get merged.

**Comment by mashrukk** [comment]
@adamperkowski Oh I see, i'll wait

**Review by adamperkowski — APPROVED** [review]

**Comment by nnyyxxxx** [comment]
I don't agree with this at all

**Comment by nnyyxxxx** [comment]
These should be kept separate so they can be maintained better.

**Comment by nnyyxxxx** [comment]
Having more than 2 scripts merged into 1 is yuck imo

**Comment by nnyyxxxx** [comment]
if you're going to do it this way than you would have to add literally every terminal install script into this that we add into linutil in the future, eventually this will become unusable and unreadable... It's best to keep expanding install scripts out of 1 file. Now this is not the same for my rewrite of the aur helper functionality #382 This PR only includes 2 aur_helpers and is very minimal due to that its supposed to be deployed anywhere in any script for Arch Linux users as a way to use their AUR Helpers when necessary.

**Comment by mashrukk** [comment]
@nnyyxxxx right now, only two terminals are being used. Alacrity and Kitty setup covers 90% of the population. I don't think Chris will add konsole or gnome terminal setup here. I understand your point, to each its own. Please check the discord message

**Comment by nnyyxxxx** [comment]
@mashrukk I still don't agree that the terminal scripts should be combined into one script; that is bad and as new terminals get added it will make this PR not good and those terminals will likely have to get added in as well, this will only make the code become unmaintainable over time and also keeping things in separate scripts is better for readability and management..

**Pull request closed without merging**